### PR TITLE
add jest config and sample test file

### PR DESCRIPTION
**Pull request by njogz**
# Description

Add jest configuration and one unit test

medic/interoperability#33

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: mediator/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+  clearMocks: true,
+  collectCoverage: true,
+  coverageDirectory: 'coverage',
+  coverageProvider: 'v8',
+  collectCoverageFrom: [
+    '<rootDir>/src/**/*.ts',
+  ],
+};
```
Comment (lorerod) [inline mediator/jest.config.js:11]: Having this file inside a /mediator/tests/ directory is better.

File: mediator/src/utils/patient.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      {
+        use: 'official',
+        family: ['Doe'],
+        given: ['John Doe']
+      }
+    ],
+    gender: 'male',
+    birthDate: '200 0-01-01'
+  });
+});
```
Comment (lorerod) [inline mediator/src/utils/patient.spec.ts:32]: Like my previous comment, having this file inside a /mediator/tests/ directory is better.

**Review by lorerod — CHANGES_REQUESTED** [review]
Thanks, @njogz for this initiative. 
I added a suggestion for the test files that may be helpful. Let me know what you think.
Also, for curiosity, why did you choose jest instead of mocha, as in cht-core?

File: mediator/src/utils/patient.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { genereateFHIRPatientResource } from "./patient";
+
+test("genereateFHIRPatientResource produces the correct format", () => {
```
Comment (andrablaj) [inline mediator/src/utils/patient.spec.ts:3]:
```suggestion
test("generateFHIRPatientResource produces the correct format", () => {
```

File: mediator/src/utils/patient.spec.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { genereateFHIRPatientResource } from "./patient";
```
Comment (andrablaj) [inline mediator/src/utils/patient.spec.ts:1]:
```suggestion
import { generateFHIRPatientResource } from "./patient";
```

File: mediator/src/utils/patient.spec.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+test("genereateFHIRPatientResource produces the correct format", () => {
+  const chtPatient = {
+    id: "123",
+    _id: "456",
+    name: "John Doe",
+    sex: "male",
+    date_of_birth: "200 0-01-01"
+  };
+
+  const FHIRPatient = genereateFHIRPatientResource(chtPatient);
```
Comment (andrablaj) [inline mediator/src/utils/patient.spec.ts:12]:
```suggestion
  const FHIRPatient = generateFHIRPatientResource(chtPatient);
```

File: mediator/src/utils/patient.spec.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { genereateFHIRPatientResource } from "./patient";
+
+test("genereateFHIRPatientResource produces the correct format", () => {
+  const chtPatient = {
+    id: "123",
+    _id: "456",
+    name: "John Doe",
+    sex: "male",
+    date_of_birth: "200 0-01-01"
```
Comment (andrablaj) [inline mediator/src/utils/patient.spec.ts:9]: I had to change it to: 
```suggestion
    date_of_birth: "2000-01-01"
```

File: mediator/src/utils/patient.spec.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    ],
+    name: [
+      {
+        use: 'official',
+        family: ['Doe'],
+        given: ['John Doe']
+      }
+    ],
+    gender: 'male',
+    birthDate: '200 0-01-01'
```
Comment (andrablaj) [inline mediator/src/utils/patient.spec.ts:30]: I had to change it to: 
```suggestion
    date_of_birth: "2000-01-01"
```

Comment (andrablaj) [inline mediator/src/utils/patient.spec.ts:30]: The unit test is failing with:

> 
> expect(received).toEqual(expected) // deep equality
> 
>     - Expected  - 1
>     + Received  + 1
> 
>     @@ -1,7 +1,7 @@
>       Object {
>     -   "birthDate": "2000-01-01",
>     +   "birthDate": "2000-01-01T00:00:00.000Z",

**Review by andrablaj — CHANGES_REQUESTED** [review]
Thanks @njogz for adding the first test! 

I committed an update to the troubleshooting page and shared some comments as the test failed on my side.

File: mediator/README.md [hunk 8]
```diff
@@ -14,4 +14,5 @@ docker run --network <openhim-network-name> --name mediator -rm -p 5005:5005 med
 
 ## Local development
 1. Run `npm install` to isntall the dependencies.
```
Comment (andrablaj) [inline mediator/README.md:16]:
```suggestion
1. Run `npm install` to install the dependencies.
```

File: mediator/README.md [hunk 9]
```diff
@@ -14,4 +14,5 @@ docker run --network <openhim-network-name> --name mediator -rm -p 5005:5005 med
...
-1. Run `npm dev` to start the mediator with watch mode. All code is in the `src` folder. The `dist` folder is generated by the build process.  
\ No newline at end of file
+1. Run `npm dev` to start the mediator with watch mode. All code is in the `src` folder. The `dist` folder is generated by the build process.
+1. Run `npm test` to run the tests.
```
Comment (andrablaj) [inline mediator/README.md:18]:
```suggestion
1. Navigate to `/mediator` folder. Run `npm test` to run the tests.
```

**Review by andrablaj — CHANGES_REQUESTED** [review]
Added a few minor suggestions. Otherwise looks good, to start with.

**Review by andrablaj — APPROVED** [review]

**Review by lorerod — APPROVED** [review]
Thanks @njogz

**Pull request merged**